openquake/hazardlib/gsim/sharma_2009.py

```diff
@@ -0,0 +1,232 @@
+
+        # Since sigma is not present in equation (1) it is not 
+        # entirely clear what it represents. For now just pass the 
+        # raw value on through.       
+
+        return mean, stddevs
+
+    def _get_stddevs(self, C, stddev_types, num_sites):
+        """
+        Return total sigma as reported in Table 2, p. 1202.
+        """
+        stddevs = []
+        for stddev_type in stddev_types:
+            assert stddev_type in self.DEFINED_FOR_STANDARD_DEVIATION_TYPES
+            stddevs.append(C['sigma'] + np.zeros(num_sites))
+        return np.array(stddevs)
+
+    def _compute_magnitude(self, rup, C):
+        """
+        Compute first two terms of equation (1) on p. 1200:
+
+        ``b1 + b2 * M``
+        """
+        return C['b1'] + C['b2']*rup.mag
+
+    def _compute_distance(self, dists, C):
+        """
+        Compute third term of equation (1) on p. 1200:
+
+        ``b3 * log(sqrt(Rjb ** 2 + b4 ** 2))``
+        """
+        return C['b3']*np.log10(np.sqrt(dists.rjb**2. + C['b4']**2.))
+
+    def _get_site_amplification(self, sites, C):
+        """
+        Compute fourth term of equation (1) on p. 1200:
+
+        ``b5 * S``
+        """
+        S = self._get_site_type_dummy_variables(sites)
+        return C['b5']*S
+        
+    def _get_mechanism(self, rup, C):
+        """
+        Compute fifth term of equation (1) on p. 1200:
+
+        ``b6 * H``
+        """
+        H = self._get_fault_type_dummy_variables(rup)
+        return C['b6']*H
+        
+    def _get_site_type_dummy_variables(self, sites):
+        """
+        Binary rock/soil classification dummy variable, ``S`` based 
+        on sites.vs30. Theshhold is not explicitly given in Sharma 
+        et al. (2009) so a value of 760 m/s was chosen. This is 
+        consistent with 
+        :mod:`openquake.hazardlib.gsim.atkinson_boore_2003`, 
+        corresponds to NEHRP class A/B, and is close to the 
+        threshhold for Eurocode 8 Class 8 (800 m/s).
+        """
+        Sab = np.array(sites.vs30 > 760.)
+        return Sab
+
+    def _get_fault_type_dummy_variables(self, rup):
+        """ Get fault-type classification dummy variable, ``H``, 
+        based on rup.rake. 
+        
+        Rake threshhold of 30 degrees is used, same as 
+        :mod:`openquake.hazardlib.gsim.boore_atkinson_2008` and 
+        :mod:`openquake.hazardlib.gsim.campbell_bozorgnia_2008`. 
+        Contrast with 45 degree threshhold used by 30 degree 
+        threshhold used in :mod:`openquake.hazardlib.gsim.zhao_2006`. 
+
+       :raises ValueError:
+            If mechanism is determined to be normal faulting, since
+            as summarized in Table 2 on p. 1197 the data used for 
+            regression included only reverse and stike-slip events.
+        """
+       
+        rake_thresh = 30.
+        
+        # normal faulting
+        Fn = np.array(rake_thresh < -rup.rake < (180. - rake_thresh))
+        
```

Comment:
Agreed. I was following the handy guide at https://github.com/gem/oq-hazardlib/wiki/Implementing-a-new-GSIM-in-the-OpenQuake-hazard-library-%5Bprovisional%5D until the very last step: pylint. Pylint had the same criticism, and a few others.

In other news, just after I submitted this push request, Professor Sharma replied to my request for an independent implementation. I how have one; and I discovered that I'd implemented "H" incorrectly - partly because of this cryptic naming!